File: resthub-apps/booking/booking-js/src/main/webapp/dao/hotel.dao.js
define([ 'resthub.dao' ], function(Dao) {

	Dao.extend("HotelDao", {

		root : 'api/hotel/',

		find : function(callback, val, page, size) {
			var url = this.root + 'search?page=' + page + '&size=' + size;
			if (val) {
				url = url + '?q=' + val;
			}
			this._get(url, callback);
		}
	}, {});
});

File: resthub-apps/booking/booking-core/src/main/java/org/resthub/booking/dao/HotelDao.java
package org.resthub.booking.dao;

import org.resthub.booking.model.Hotel;
import org.resthub.core.dao.GenericDao;
import org.synyx.hades.domain.Page;
import org.synyx.hades.domain.Pageable;

/**
 * @author Guillaume Zurbach
 * @author bmeurant <Baptiste Meurant>
 */
public interface HotelDao extends GenericDao<Hotel, Long> {

	/**
	 * Find hotel by fulltext search.
	 * 
	 * @param query
	 *            query in HibernateSerach (Lucene) format
	 * @param pageable
	 *            pagination context
	 * @return hotels matching query or null if query string is in an incorrect
	 *         format.
	 */
	Page<Hotel> find(String query, Pageable pageable);

	/**
	 * Rebuil full index (Hibernate Search related).
	 */
	void rebuildIndex();
}
